clang/test/dpct/helper_files_ref/include/rng_utils.hpp

```diff
@@ -174,6 +176,235 @@ template <typename engine_t> class rng_generator {
+  if (seed == _seed) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    _philox4x32x10_engine = std::make_shared<oneapi::mkl::rng::philox4x32x10>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine = std::make_shared<oneapi::mkl::rng::mrg32k3a>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine = std::make_shared<oneapi::mkl::rng::mt19937>(
+        dpct::cpu_device().default_queue(), seed);
+    break;
+  default:
+    break;
+  }
+  return 0;
+}
+
+int set_generator_dimensions(const std::uint32_t &dimensions) {
+  if (dimensions == _dimensions) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::sobol:
+    _sobol_engine = std::make_shared<oneapi::mkl::rng::sobol>(
+        dpct::cpu_device().default_queue(), dimensions);
+    break;
+  default:
+    break;
+  }
+  return 0;
+}
+
+int set_generator_queue(const sycl::queue &queue) {
+  if (queue == _queue) {
+    return 0;
+  }
+  destroy_generator();
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    _philox4x32x10_engine =
+        std::make_shared<oneapi::mkl::rng::philox4x32x10>(queue, _seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine =
+        std::make_shared<oneapi::mkl::rng::mrg32k3a>(queue, _seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(queue, _seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine = std::make_shared<oneapi::mkl::rng::mt19937>(queue, _seed);
+    break;
+  case random_engine_type::sobol:
+    _sobol_engine =
+        std::make_shared<oneapi::mkl::rng::sobol>(queue, _dimensions);
+    break;
+  }
+  return 0;
+}
+
+int destroy_generator() {
+  _philox4x32x10_engine.reset();
+  _mrg32k3a_engine.reset();
+  _mt2203_engine.reset();
+  _mt19937_engine.reset();
+  _sobol_engine.reset();
+  return 0;
+}
+
+int generate(unsigned int *outputPtr, std::int64_t n) {
+  oneapi::mkl::rng::uniform_bits<uint32_t> distr;
+  return generate(distr, (uint32_t *)outputPtr, n);
+}
+
+int generate_long_long(unsigned long long *outputPtr, std::int64_t n) {
+  oneapi::mkl::rng::uniform_bits<uint64_t> distr;
+  return generate(distr, (uint64_t *)outputPtr, n);
+}
+
+int generate_log_normal(float *outputPtr, std::int64_t n) {
+  oneapi::mkl::rng::lognormal<float> distr;
+  return generate(distr, outputPtr, n);
+}
+
+int generate_log_normal_double(double *outputPtr, std::int64_t n) {
+  oneapi::mkl::rng::lognormal<double> distr;
+  return generate(distr, outputPtr, n);
+}
+
+int generate_normal(float *outputPtr, std::int64_t n) {
```

Comment:
Let's use naming consistent with oneMKL
```suggestion
int generate_gaussian(float *outputPtr, std::int64_t n) {
```

Reply:
Fixed.